inc/init.php

```diff
@@ -495,7 +495,11 @@ function getBaseURL($abs = null)
 
     //split hostheader into host and port
     if (isset($_SERVER['HTTP_HOST'])) {
-        $parsed_host = parse_url('http://' . $_SERVER['HTTP_HOST']);
+        $cur_host = $_SERVER['HTTP_HOST'];
+        if (isset($_SERVER['HTTP_X_FORWARDED_HOST']) && preg_match('/'.$conf['trustedproxy'].'/', $_SERVER['REMOTE_ADDR'])) {
```

Comment:
You also need to check for `empty($conf['trustedproxy'])`.

Reply:
Well, if it is empty, that just means every remote address matches the expression. In that case, this is a deliberate setting from the administrator, right? Or do you consider an empty value indicates no proxy is trusted?